carp.data.core/src/commonMain/kotlin/dk/cachet/carp/data/infrastructure/DataStreamServiceLog.kt

```diff
@@ -0,0 +1,48 @@
+package dk.cachet.carp.data.infrastructure
+
+import dk.cachet.carp.common.application.UUID
+import dk.cachet.carp.common.infrastructure.services.ApplicationServiceLog
+import dk.cachet.carp.common.infrastructure.services.LoggedRequest
+import dk.cachet.carp.common.infrastructure.services.SingleThreadedEventBus
+import dk.cachet.carp.data.application.DataStreamBatch
+import dk.cachet.carp.data.application.DataStreamId
+import dk.cachet.carp.data.application.DataStreamService
+import dk.cachet.carp.data.application.DataStreamsConfiguration
+
+
+/**
+ * A proxy for a data stream [service] which notifies of incoming requests and responses through [log]
+ * and keeps a history of requests in [loggedRequests].
+ */
+class DataStreamServiceLog(
```

Comment:
I think this really convinces me that it should be called `DataStreamServiceLoggingProxy` :)